gui: track when we've caught up (#6041)
firedancer-io/firedancer: book/api/websocket.md

validators fork choice. A distance of more than 150 means the validator
is considered delinquent.

#### `summary.slot_max_known`
| frequency       | type     | example |
|-----------------|----------|---------|
| *Once* + *Live* | `number` | `100`   |

The largest slot that is known by the validator to have been published

> to the blockchain.  This is typically going to be the largest slot we've

Reviewer: Markdown is single spaced

seen in a received turbine shred, but can also be a slot for which we
were just leader.  During boot, the max known slot may not be known yet
if we haven't received any shreds.  In this case this message will
publish `null`.

It is worth nothing that slot contained might be momentarily arbitrarily
larger than the ground truth.  If this happens, it will return to an
accurate estimate after about 4.8 seconds.  This is because the slot

> number is obtained the header on incoming shreds.  In the worst case, a

Reviewer: ```suggestion
number is derived from the header on incoming shreds.  In the worst case, a
```

malicious leader shred can contain an arbitrarily large slot on a fork

> that that chain wont actually converge on.  During catchup validator has

Reviewer: ```suggestion
that that chain wont actually converge on.  During catchup the validator has
```

maybe just do a small pass on this whole comment

no way of knowing that this fork is incorrect.  After 4.8 seconds of
receiving a malicious slot number, it will be forgotten, which is how
the node ensures this message is eventually consistent.

> NOTE: this message is only supported on the firedancer client, the

Reviewer: ```suggestion
NOTE: this message is only supported on the Firedancer client, the
```

always capitalize

frankendancer client will always publish `null` for this message

#### `summary.slot_caught_up`
| frequency       | type     | example |
|-----------------|----------|---------|

> | *Once* + *Live* | `number` | `100`   |

Reviewer: The type here is nullable, which is `number|null` per typescript. See other types in the file
